Controller/AdminResettingController.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/*
+ * This file is part of the Sonata package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Controller;
+
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use FOS\UserBundle\Controller\ResettingController;
+use FOS\UserBundle\Model\UserInterface;
+
+/**
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+class AdminResettingController extends ResettingController
+{
+    /**
+     * Request reset user password: show form
+     */
+    public function requestAction()
+    {
+        if ($this->container->get('security.context')->isGranted('IS_AUTHENTICATED_FULLY'))
+        {
+            return new RedirectResponse($this->container->get('router')->generate('sonata_admin_dashboard'));
+        }
+
+        return $this->container->get('templating')->renderResponse('SonataUserBundle:Admin:Security/Resetting/request.html.'.$this->getEngine(), array(
+            'base_template' => $this->container->get('sonata.admin.pool')->getTemplate('layout'),
+            'admin_pool'    => $this->container->get('sonata.admin.pool'),
+        ));
+    }
+
+    /**
+     * Request reset user password: submit form and send email
+     */
+    public function sendEmailAction()
+    {
+        $username = $this->container->get('request')->request->get('username');
+
+        /** @var $user UserInterface */
+        $user = $this->container->get('fos_user.user_manager')->findUserByUsernameOrEmail($username);
+
+        if (null === $user) {
+            return $this->container->get('templating')->renderResponse('SonataUserBundle:Admin:Security/Resetting/request.html.'.$this->getEngine(), array(
+                'invalid_username' => $username,
+                'base_template'    => $this->container->get('sonata.admin.pool')->getTemplate('layout'),
+                'admin_pool'       => $this->container->get('sonata.admin.pool'),
+            ));
+        }
+
+        if ($user->isPasswordRequestNonExpired($this->container->getParameter('fos_user.resetting.token_ttl'))) {
+            return $this->container->get('templating')->renderResponse('SonataUserBundle:Admin:Security/Resetting/passwordAlreadyRequested.html.'.$this->getEngine(), array(
+                'base_template'    => $this->container->get('sonata.admin.pool')->getTemplate('layout'),
+                'admin_pool'       => $this->container->get('sonata.admin.pool'),
+            ));
+        }
+
+        if (null === $user->getConfirmationToken()) {
+            /** @var $tokenGenerator \FOS\UserBundle\Util\TokenGeneratorInterface */
```

Comment:
It is from the original FOSUserBundle code. I don't mind. I guess it is mainly there for IDE purposes.